pkg/corerp/backend/deployment/deploymentprocessor.go

```diff
@@ -575,17 +574,16 @@ func (dp *deploymentProcessor) getResourceDataByID(ctx context.Context, resource
 
 func (dp *deploymentProcessor) buildResourceDependency(resourceID resources.ID, applicationID string, resource v1.DataModelInterface, outputResources []rpv1.OutputResource, computedValues map[string]any, secretValues map[string]rpv1.SecretValueReference, recipeData portableresources.RecipeData) (ResourceData, error) {
 	var appID *resources.ID
+	// Application id is mandatory for core resource types and is a required field.
```

Comment:
minor update since some resources under Applications.Core namespace do not require application id like SecretStore and Extender.

```suggestion
	// Application id is mandatory for some of the core resource types and is a required field.
```